mllib/src/main/scala/org/apache/spark/mllib/classification/NaiveBayes.scala

```diff
@@ -65,6 +67,70 @@ class NaiveBayesModel private[mllib] (
   override def predict(testData: Vector): Double = {
     labels(brzArgmax(brzPi + brzTheta * testData.toBreeze))
   }
+
+  override def save(sc: SparkContext, path: String): Unit = {
+    val sqlContext = new SQLContext(sc)
+    import sqlContext._
+
+    // Create JSON metadata.
+    val metadataRDD =
+      sc.parallelize(Seq((this.getClass.getName, formatVersion))).toDataFrame("class", "version")
+    metadataRDD.toJSON.repartition(1).saveAsTextFile(path + "/metadata")
+
+    // Create Parquet data.
+    val data = NaiveBayesModel.Data(labels, pi, theta)
+    val dataRDD: DataFrame = sc.parallelize(Seq(data))
+    dataRDD.repartition(1).saveAsParquetFile(path + "/data")
+  }
+
+  override protected def formatVersion: String = NaiveBayesModel.formatVersion
+
+}
+
+object NaiveBayesModel extends Importable[NaiveBayesModel] {
+
+  /** Model data for model import/export */
+  private case class Data(labels: Array[Double], pi: Array[Double], theta: Array[Array[Double]])
```

Comment:
This `Data` class should live inside `ImporterV1`. If `ImporterV2` uses a different format, it is hard to update `Data` if it is global to both importers.